### PR TITLE
Ionization

### PR DESCRIPTION
**Pull request by KhalilBryant**
- [x] I have added a changelog entry for this pull request.

- [x] If adding new functionality, I have added tests and
      docstrings.

- [x] I have fixed any newly failing tests.

This adds the `ionization` module to the formulary as well as Z_bal from the Saha equation

### CODE REVIEW

File: plasmapy/formulary/dimensionless.py [hunk 1]
```diff
@@ -171,11 +171,10 @@ def Mag_Reynolds(U: u.m / u.s, L: u.m, sigma: u.S / u.m) -> u.dimensionless_unsc
...
     .. math::
 
-        Rm = \frac{U L}{\\eta}
+        Rm = \frac{U L}{\eta}
 
-        where: \eta = \frac{1}{\mu_0 \sigma}
-
-        and \mu_0 is the permeability of free space.
+    where :math:`\eta = \frac{1}{\mu_0 \sigma}`
+    and :math:`\mu_0` is the permeability of free space.
```
Comment (StanczakDominik) [inline plasmapy/formulary/dimensionless.py:177]: It definitely looks like something is off with your git tree history... These changes should really not be showing up. Could you try merging the current master branch?

File: plasmapy/formulary/ionization.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+from plasmapy.utils.decorators import (
+    angular_freq_to_hz,
+    check_relativistic,
+    validate_quantities,
+)
+from astropy.constants import c, a0, k_B
+from numpy import pi, exp, sqrt, log
+
+
+def Z_bal(n: u.m ** -3, T_e: u.K):
```
Comment (StanczakDominik) [inline plasmapy/formulary/ionization.py:15]: This definitely needs better naming... Something balanced, where the something is...?

File: plasmapy/formulary/ionization.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    r"""
+    Z_bal is derived from the Saha equation with the assumptions that
```
Comment (StanczakDominik) [inline plasmapy/formulary/ionization.py:17]: Likewise, here we need to describe what it is this calculates, and only then can we state the assumptions that go into it.

File: plasmapy/formulary/ionization.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        The electron number density of the plasma.
+    Returns
+    -------
+    Z: `~astropy.units.Quantity`
+        The average ionization state of the ions in the plasma
+        assuming that the number of ions in each state are equal.
+
+    """
+    E_H = 13.6 * u.eV
+
```
Comment (StanczakDominik) [inline plasmapy/formulary/ionization.py:50]: Not to complain, but now I'm curious whether this would not be in astropy constants or in units as 1 Rydberg?

Comment (KhalilBryant) [reply plasmapy/formulary/ionization.py:50]: I just looked and it is, I can change that

**Comment by StanczakDominik** [comment]
Remember to commit/push the alias change from today, and this will probably be a quick merge! :)

**Comment by rocco8773** [comment]
touches issue #856

**Pull request merged**